woocommerce/woocommerce-admin-test-helper
# Add woocommerce com url overwrite and ability to clear transients

Allows us to overwrite the `woocommerce.com` url for testing without having to modify our code.
Also allows us to clear the payment and free extensions spec transients that come from woocommerce.com.

### Screenshot

<img width="1364" alt="Screen Shot 2021-08-13 at 11 44 16 AM" src="https://user-images.githubusercontent.com/2240960/129374725-094b0860-90ca-4276-ab40-488415b1ea1d.png">

### Testing instructions

**It will help if you already have a woocommerce.com environment set up locally**

1. Make sure your local woocommerce.com environment is running
2. Navigate to the business details step in the onboarding wizard, or the payment task (both use woocommerce.com)
3. In a separate tab open up **Tools > WCA Test Helper > Tools**
4. The clear transient tool should include one or two items in the dropdown (depending if you loaded the payments task or the business details step), you should be able to clear them and they will disappear out of the dropdown
5. Now update the `woocommerce.com` url to whatever you use locally (default: `http://woocommerce.test`), it should say `Disabled`
6. Click `Toggle`, after updating it should say `Enabled`
7. Go back to the other tab with the (payments task or business details step) and refresh the page
8. It should now load data from the `woocommerce.com` local environment, you can see this more easily on the payments task to see where the payment images came from (they should come from your local).
9. Go back to the other tab and Toggle the overwrite again (disabling it) also refresh the page and clear the transients again.
10. Go back to the payments tab again and refresh, it should be retrieving from `woocommerce.com` again.
